### ISSUE TITLE
[10.0.0-beta.2] Dom order incorrect after state update

### ISSUE THREAD

**Issue by tholman** [preactjs/preact#1699]
Again, related to https://github.com/preactjs/preact/issues/1635 and  https://github.com/preactjs/preact/issues/1632 ... but felt slightly different, so a new ticket here.

Here's a demo of the broken case https://codesandbox.io/s/k0jiy

In this instance, the first component is

```
<Fragment>
  {false && <p>Never exists</p>}
  <p>First Component</p>
</Fragment>
```

After the state updates the second component jumps the first. I think there could be a few things at play here:

- If the false condition is moved below the `<p>` tags, the second component will not jump the first
- If the false conditional is removed completely, it will not jump the first
- If the `Fragment` is replaced with a div, it will not jump the first

**Comment by JoviDeCroock** [comment]
I'll make a test for this case and see if the PR covers it, thanks for the reproduction!

**Comment by marvinhagemeister** [comment]
@andrewiggins added a test case for that in #1700 🎉

**Comment by marvinhagemeister** [comment]
This is fixed with #1689 and #1700 :tada:
